src/argilla/server/security/auth_provider/db/provider.py

```diff
@@ -52,36 +54,31 @@ async def login_for_access_token(
             user = await accounts.authenticate_user(db, form_data.username, form_data.password)
             if not user:
                 raise UnauthorizedError()
+
             access_token_expires = timedelta(minutes=self.settings.token_expiration_in_minutes)
             access_token = self._create_access_token(user.username, expires_delta=access_token_expires)
+
             return Token(access_token=access_token)
 
-    def _create_access_token(self, username: str, expires_delta: Optional[timedelta] = None) -> str:
-        """
-        Creates an access token
+    async def get_current_user(
+        self,
+        security_scopes: SecurityScopes,
+        request: Request,
+        db: AsyncSession = Depends(get_async_db),
+        api_key: Optional[str] = Depends(api_key_header),
+        token: Optional[str] = Depends(_oauth2_scheme),
+    ) -> User:
+        user = None
 
-        Parameters
-        ----------
-        username:
-            The user name
-        expires_delta:
-            Token expiration
+        if api_key:
+            user = await accounts.get_user_by_api_key(db, api_key)
+        elif token:
+            user = await self.fetch_token_user(db, token)
```

Comment:
To be a little bit more clear about the flow

```suggestion
        if api_key:
            user = await accounts.get_user_by_api_key(db, api_key)
        elif token:
            user = await self.fetch_token_user(db, token)
        else:
            raise UnauthorizedError()
```

With this you can delete the previous lines with:

```python
user = None
```

And also delete:

```python
if user is None:
    raise UnauthorizedError()
```

But still do the `return user` at the end.

Reply:
Not really, both methods `accounts.get_user_by_api_key` and `self.fetch_token_user` can return `None` if the user is not found by provided info. We should refactor first those methods to apply your suggestion. I will skip for the moment.